client/src/features/shell/pages/authenticated-layout.tsx

```
@@ -0,0 +1,463 @@
+  component: AuthenticatedLayout,
+  beforeLoad: ({ location }) => {
+    const { isAuthenticated } = useAuthStore.getState();
+    if (!isAuthenticated) {
+      // Redirect to login with redirect param
+      throw redirect({
+        to: '/login',
+        search: { redirect: location.pathname },
+        replace: true,
+      });
+      // Returning false prevents the route from loading
+      return false;
```

Biome (1.9.4)

[error] This code is unreachable

... because this statement will throw an exception beforehand


(lint/correctness/noUnreachable)